lua/nvim_lsp/pyls_ms.lua

```diff
@@ -0,0 +1,133 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local lsp = vim.lsp
+
+local name = "pyls_ms"
+
+local function get_python_version()
+  local f = io.popen("python --version 2>&1") -- runs command
+  local l = f:read("*a") -- read output of command
+  f:close()
+  return l:match("^Python%s*(...).*%s*$")
+end
+
+local function make_installer()
+  local P = util.path.join
+  local install_dir = P{util.base_install_dir, name}
+
+  local bin = P{install_dir, "Microsoft.Python.LanguageServer.dll"}
+  local cmd = {"dotnet", "exec", bin}
+
+  local X = {}
+  function X.install()
+    local install_info = X.info()
+    if install_info.is_installed then
+      print(name, "is already installed")
+      return
+    end
+    if not (util.has_bins("curl")) then
+      error('Need "curl" to install this.')
+      return
+    end
+    if not util.has_bins("sh", "chmod", "unzip") then
+      error('Need the binaries "sh", "chmod", "unzip" to install this')
+      return
+    end
+    local script = [=[
+set -e
+# Install dotnet if not available.
+which dotnet  >/dev/null || {
+  test -x dotnet || {
+    curl -L https://dot.net/v1/dotnet-install.sh | sh
+  }
+  export PATH=$PWD:$PATH
+}
```

Comment:
I'm not sure it's reasonable for us to install dotnet, that would be like installing python. I would suggest showing an error and mention `curl -L https://dot.net/v1/dotnet-install.sh | sh` in the docs.

Reply:
Fixed